### PR TITLE
Convert tests to java8 runtime

### PR DESCRIPTION
**Pull request by elharo**
@briandealwis

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ProjectContextMenuTest.java [hunk 1]
```diff
@@ -58,25 +58,6 @@ public void testPlainJavaProject() {
         not(MenuMatcher.hasMenuItem("Reconfigure for App Engine Java 8 runtime")));
   }
 
-  @Test
-  public void testDynamicWebProjectJava7() {
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ProjectContextMenuTest.java]: This test should be kept — we should be able to convert a Java 7 web project to App Engine Java 8

File: plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ProjectContextMenuTest.java [hunk 2]
```diff
@@ -96,24 +77,6 @@ public void testDynamicWebProjectJava8() {
         not(MenuMatcher.hasMenuItem("Reconfigure for App Engine Java 8 runtime")));
   }
 
-  @Test
-  public void testAppEngineStandardJava7() {
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ProjectContextMenuTest.java]: This test should be kept until we remove App Engine Java 7

Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ProjectContextMenuTest.java]: done

File: plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ProjectContextMenuTest.java (hunk 1 above)
Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ProjectContextMenuTest.java]: done

**Comment by briandealwis** [comment]
Tests are failing as `JRE8` is defined in a separate bundle.
```
Caused by: java.lang.IllegalArgumentException: Version JRE8 of project facet com.google.cloud.tools.eclipse.appengine.facets.standard does not exist.
        at org.eclipse.wst.common.project.facet.core.util.internal.Versionable.getVersion(Versionable.java:80)
        at org.eclipse.wst.common.project.facet.core.internal.ProjectFacet.getVersion(ProjectFacet.java:1)
        at com.google.cloud.tools.eclipse.appengine.facets.AppEngineStandardFacet.<clinit>(AppEngineStandardFacet.java:59)
        ... 55 more
```

We can annotate the various `*.test/pom.xml` with requiring .standard.java8, but I suspect a slightly easier way is to use the `Require-Capability`/`Provide-Capability`.

**Comment by briandealwis** [comment]
More specifically, I think we'd add to `.standard.java8`:
```
Provide-Capability: com.google.cloud.tools.facets.appengine;type:List="jre8"
```

and the `.test` fragments would then need:
```
Require-Capability: com.google.cloud.tools.facets.appengine;filter:="(type=jre8)"
```

**Pull request merged**